tas2781-fix.sh

```diff
@@ -280,6 +282,14 @@ execute_fix() {
 
     i2cset -f -y $i2c_bus $value 0x00 0x00 # Page 0x00
     i2cset -f -y $i2c_bus $value 0x02 0x00 # Play audio, power up with playback, IV enabled
+
+    if [ $balance_index -eq 0 ]; then # Set left/right balance
```

Comment:
Line 264 already sets the left/right channel based on the existing i2c configuration. I prefer doing it in a way that doesn't make assumptions about the order of the left/right components. See line 245, where the existing bits representing the r/l channels are cached in a variable. Have you verified that the fix doesn't work without this newer addition?

Reply:
Yeah, the code on main doesn't work for me. The i2cget that's run to build the address_channels array just ends up putting getting 0/0x0a for both addresses. This results in just the left speaker working. In order for my speakers to both function, I need to set 32/0x2e on address 61 (the 2nd address on my machine after 56).

I'm not sure why it doesn't get the proper value. I cribbed this method of manually setting 0x2e on the 2nd address from one of these scripts floating around from one of the TI devs, but it's been months since I was messing with that so I don't remember where exactly I found it. They used this same method of keeping count and running a modulo 2 on the count to determine which value to set--just a brute force fix.

I'm not sure if there's another programmatic way to get the proper value, and if not, if you'd like me to write in some kind of exception case for this 9i configuration?

Reply:
Oh and if I try to set 32 on address 61, as the value would be formatted by the xargs command, it makes the speaker just hiss super loud blown out static. I have to set it with the actual value of 0x2e for to work.

Reply:
The mod 2 solution has worked for me in the past too, so I'm fine with using that, given it's proven more reliable. The change I'd recommend then is:
- remove the logic that fetches and caches the channel configuration as it's no longer needed
- move your change up to line 264, replacing the old assignment to 0x0a

I can go ahead and see if there's a way to more reliably detect l/r settings in the meantime.

Reply:
Changes have been made--it seems to be working well for me so far.